### ISSUE TITLE
regex check of mc options needs to account for {} in values - URGENT!

### ISSUE THREAD

**Issue by berryld1** [vubiostat/redcapAPI#138]
Error in grepl(sprintf("^(%s)$", paste0(c(coding, names(coding)), collapse = "|")),  : 
  invalid regular expression '^(1|2|3|4|5|6|7|8|Woman|Man|Transgender woman or transfeminine|Transgender man or transmasculine|Non-binary|Genderqueer|Two-spirit|Something else, please specify: {q1_genderother})$', reason 'Invalid contents of {}'
In addition: There were 13 warnings (use warnings() to see them)

**Comment by berryld1** [comment]
I cannot find where this is run:
grepl(sprintf("^(%s)$", paste0(c(coding, names(coding)), collapse = "|"))

I'm pretty sure I've looked through all the code...

Anyhow, this ought to fix it:
grepl(sprintf("^(%s)$", paste0(c(coding, gsub("\\}", "\\\\}", gsub("\\{", "\\\\{", names(coding)))), collapse = "|"))

When this renders, the backslashes are escaping themselves...

For clarify: gsub(two-backslashes-{, four-backslashes-{, string)

**Comment by obregos** [comment]
I've replicated this error using the Basic Data Types (For Testing) PID 167805. On instrument `multiple_choice` I added a field `embedded_choice_test` that has an option for existing field `radio_test`.

Running `exportRecordsTyped(rcon)` on this project produces the original error.

Updating to `valChoice <- function(x, field_name, coding) x %in% coding | x %in% names(coding)` works to remove the error message and formats the output as expected. 

One thing I noticed: If one of the 'Other' options from the embedded `radio_test` is selected for `embedded_choice_test` but the 'Other' radio option for `embedded_choice_test` is not selected then the field is exported as NA which is not correct.
